tests/unit/testsuite/Bolt/Boltpay/controllers/ShippingControllerTest.php

```diff
@@ -2,69 +2,318 @@
+            Bolt_Boltpay_ProductProvider::deleteDummyProduct($productId);
+        }
+
+        Mage::app()->getCache()->clean('matchingAnyTag', array('BOLT_QUOTE_PREFETCH'));
+    }
+
+    /**
+     * Test to see if cache is in a valid state after prefetch data is sent.  Prior to
+     * the prefetch, there should be no cache data.  After the prefetch, there should be
+     * cached data.
+     *
+     * @throws ReflectionException      on unexpected problems with reflection
+     * @throws Zend_Cache_Exception     on unexpected problems reading or writing to Magento cache
+     */
+    public function testIfEstimateIsCachedAfterPrefetch() {
+
+        Mage::app()->getCache()->clean('matchingAnyTag', array('BOLT_QUOTE_PREFETCH'));
+
+        /** @var Mage_Sales_Model_Quote $quote */
+        $quote = Mage::getSingleton('checkout/session')->getQuote();
+
+        $quote
+            ->removeAllAddresses()
+            ->removeAllItems()
+            ->setCustomerId(25)
+            ->setCustomerTaxClassId(3);
+
+        foreach(self::$_productIds as $productId) {
+            $this->testHelper->addProduct($productId, rand(1,3));
+        }
+
+        $quote->setTotalsCollectedFlag(false);
+        $quote->collectTotals()->save();
+
+        $reflectedShippingController = new ReflectionClass($this->_shippingController);
+        $reflectedGetEstimateCacheIdentifier = $reflectedShippingController->getMethod('getEstimateCacheIdentifier');
+        $reflectedGetEstimateCacheIdentifier->setAccessible(true);
+
+        $expectedAddressData = array(
+            'city'       => 'Beverly Hills',
+            'country_id' => 'US',
+            'region_id'  => '12',
+            'region' => 'California',
+            'postcode' => '90210'
+        );
+        $expectedCacheId = $reflectedGetEstimateCacheIdentifier->invoke($this->_shippingController, $quote, $expectedAddressData);
+        $estimatePreCall = unserialize(Mage::app()->getCache()->load($expectedCacheId));
+
+
+        $geoIpAddressData = array(
+            'city'       => 'Beverly Hills',
+            'country_code' => 'US',
+            'region_code'  => 'CA',
+            'region_name' => 'California',
+            'zip_code' => '90210'
+        );
+        $reflectedRequestJson = $reflectedShippingController->getProperty('_requestJSON');
+        $reflectedRequestJson->setAccessible(true);
+        $reflectedRequestJson->setValue($this->_shippingController, json_encode($geoIpAddressData));
+
+        $this->_shippingController->prefetchEstimateAction();
+
+        $actualAddressData = json_decode($this->_shippingController->getResponse()->getBody(), true)['address_data'];
+
+        $actualCacheId = $reflectedGetEstimateCacheIdentifier->invoke($this->_shippingController, $quote, $actualAddressData);
+        $estimatePostCall = unserialize(Mage::app()->getCache()->load($actualCacheId));
+
+        Mage::app()->getCache()->clean('matchingAnyTag', array('BOLT_QUOTE_PREFETCH'));
```

Comment:
why you need this? maybe move to setUp or tearDown method if this is a cleanup?

Reply:
removed.